packages/babel-cli/src/babel/dir.ts

```diff
@@ -177,34 +177,70 @@ export default async function ({
     // This, alongside with debounce, allows us to only log
     // when we are sure that all the files have been compiled.
     let processing = 0;
+    const { filenames } = cliOptions;
+    let getBase;
+    if (filenames.length === 1) {
+      // fast path: If there is only one filenames, we know it must be the base
+      const base = filenames[0];
+      const absoluteBase = path.resolve(base);
+      getBase = filename => {
+        return filename === absoluteBase ? path.dirname(base) : base;
+      };
+    } else {
+      // A map from absolute compiled file path to its base, from which
+      // the output destination will be determined
+      const filenameToBaseMap: Map<string, string> = new Map(
+        filenames.map(filename => {
```

Comment:
I was wondering whether these could be globs or not, but they are already resolved at https://github.com/babel/babel/blob/193728405b6efc46a828e665510a265fc72b344c/packages/babel-cli/src/babel/options.ts#L192-L197

Reply:
Yes they are resolved prior to watch setup, which means if you run `babel */src --watch` and then create a new folder `module3/src`, it will not be added to the watcher.